### PR TITLE
Surface concept schema changes via sdk query updates

### PR DESCRIPTION
**Pull request by THOR300**
# Description

- Integrates the latest vespa schema changes to accommodate concepts. 
- Integrates new test data that includes concepts. 
- Updates the sdk filtering functionality to enable concept filtering. 
- Updates the unit tests to include tests of the new concept fields. 

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [ ] Patch
- [X] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail (integrated soon)
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/climatepolicyradar/cpr-sdk/blob/main/.github/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/climatepolicyradar/cpr-sdk/blob/main/.github/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

### CODE REVIEW

File: src/cpr_sdk/models/search.py [hunk 1]
```diff
@@ -45,6 +46,42 @@ class MetadataFilter(BaseModel):
...
+
+        E.g. parent_concept_ids_flat: "Q1" should be parent_concept_ids_flat: "Q1,"
+        but you will also return "Q12, Q123," which is invalid.
+        """
+        if self.name == "parent_concept_ids_flat" and self.value[-1] != ",":
+            self.value = self.value + ","
+        return self
+
+
+class ConceptHit(BaseModel):
```
Comment (THOR300) [inline src/cpr_sdk/models/search.py]: Move this to a better location in the sdk.

File: tests/test_search_adaptors.py [hunk 2]
```diff
@@ -464,6 +466,85 @@ def test_vespa_search_adaptor__corpus_type_name(
...
+                            for hit_concept_filter_val in hit_concept_filter_vals
+                        ]
+                    )
+                else:
+                    assert any(
+                        [
+                            hit_concept_filter_val == concept_filter["value"]
+                            for hit_concept_filter_val in hit_concept_filter_vals
+                        ]
+                    )
```
Comment (THOR300) [inline tests/test_search_adaptors.py:545]: Check that the parent concept ids matches the flattened version.

File: src/cpr_sdk/models/search.py [hunk 3]
```diff
@@ -45,6 +45,71 @@ class MetadataFilter(BaseModel):
...
+            ",".join(
+                [parent_concept["name"] for parent_concept in self.parent_concepts]
+            )
+        )
+
+        if parent_concept_ids_flattened[-1] != ",":
+            parent_concept_ids_flattened += ","
+
+        if not self.parent_concept_ids_flat == parent_concept_ids_flattened:
+            raise ValueError(
```
Comment (THOR300) [inline src/cpr_sdk/models/search.py:108]: Add a test.

File: src/cpr_sdk/models/search.py [hunk 4]
```diff
@@ -45,6 +45,69 @@ class MetadataFilter(BaseModel):
...
+    A concept extracted from a passage of text.
+
+    This refers to a span of text within passage that holds a concept.
+    E.g. "Adaptation strategy" is a concept within a passage starting at index 0 and
+    ending at index 17.
+    """
+
+    id: str
+    name: str
+    parent_concepts: List[dict[str, str]]
```
Comment (THOR300) [inline src/cpr_sdk/models/search.py:86]: @olaughter suggestion: quoting instead of the commas. 

`"Q1","Q2","Q3"`

Comment (THOR300) [reply src/cpr_sdk/models/search.py:86]: Benchmarking of field searching performance with different delimiters and the use of match vs contains.

File: src/cpr_sdk/version.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 _MAJOR = "1"
 _MINOR = "8"
-_PATCH = "0"
+_PATCH = "1"
```
Comment (THOR300) [inline src/cpr_sdk/version.py]: Not a patch but minor.

**Review by olaughter — APPROVED** [review]

**Pull request merged**